### PR TITLE
[FEATURE] added order dependency workflows

### PR DESCRIPTION
**Pull request by turbo-ele**
**Tasks:**
- [x] added getWorkflows to dependency manager
- [x] javascript handling of order's workflows
- [x] gather feedback

**Informations:**

| Q | A |
| --- | --- |
| Tests pass? | - |
| Fixed tickets | - |
| BC Breaks | - |
| Doc | - |

**Requires:**
https://github.com/sulu-cmf/SuluSalesCoreBundle/pull/12

### CODE REVIEW

File: Resources/public/js/components/orders/components/form/main.js [hunk 1]
```diff
@@ -60,43 +64,34 @@ define([
...
-                } else if (this.orderStatusId === OrderStatus.CONFIRMED) {
-                    workflow.items.push(workflowItems.edit);
-                }
-
-                // more than in cart
-                if (this.orderStatusId >= OrderStatus.CONFIRMED) {
-                    workflow.items.push(workflowItems.divider);
-                    workflow.items.push(workflowItems.shipping);
+                // add workflows provided by api
+                for (i = -1, len = data.workflows.length; ++i < len;) {
```
Comment (christianbader) [inline Resources/public/js/components/orders/components/form/main.js:74]: cool guy loop isn't as cool as you think dude

File: Resources/public/js/components/orders/components/form/main.js [hunk 2]
```diff
@@ -106,6 +101,31 @@ define([
...
         /**
+         * creates a callback for a workflow
+         * @param workflow
+         */
+        createWorkflowCallback = function(workflow) {
+            // if event is defined, call event
+            if (workflow.hasOwnProperty('event') && !!workflow.event) {
+                var params = workflow.parameters || null
+                this.sandbox.emit(workflow.event, params);
+
```
Comment (christianbader) [inline Resources/public/js/components/orders/components/form/main.js]: remove empty line

File: Resources/public/js/components/orders/main.js [hunk 3]
```diff
@@ -186,6 +186,9 @@ define([
             successCallback = typeof(successCallback) === 'function' ? successCallback : null;
             failCallback = typeof(failCallback) === 'function' ? failCallback : null;
 
+            if (typeof id === 'object' && id.hasOwnProperty('id')) {
```
Comment (christianbader) [inline Resources/public/js/components/orders/main.js:190]: a comment would be nice here

**Comment by christianbader** [comment]
else ack

File: Resources/public/js/components/orders/components/form/main.js (hunk 1 above)
Comment (turbo-ele) [inline Resources/public/js/components/orders/components/form/main.js:74]: yes it is

**Pull request merged**